### PR TITLE
Implement many of the python methods for Chip

### PR DESCRIPTION
**Pull request by peter-madigan**

### CODE REVIEW

**Comment by samkohn** [comment]
I don't think the Python interface requires bit manipulation. It should all be possible with Python lists.

**Comment by peter-madigan** [comment]
I was thinking that a bit mask would be the easiest way to define a channel
map. We could just as easily use lists, but then we'd have to initialize a
list every time. I could go either way.

File: python/larpix.py [hunk 1]
```diff
@@ -16,43 +16,155 @@ def __init__(self, chip_id, io_chain):
...
 
     def set_csa_gain(self, gain):
-        pass
-
-    def enable_
+        self.configuration.csa_gain = gain
+        return 0
+
+    def enable_channels(self, map):
+        self.configuration.channel_mask = self.configuration.channel_mask & ~map
```
Comment (samkohn) [inline python/larpix.py]: Here `self.configuration.channel_mask` is a c array (defined [here](https://github.com/samkohn/larpix-control/pull/5/files#diff-bf8d555f276f235c897b5bbe727efc42R46)) so this assignment doesn't work.

**Comment by samkohn** [comment]
I had only skimmed your code before. But the bit mask code you wrote doesn't work. The `channel_mask` (and other by-channel settings) are represented in the configuration struct by c arrays, not by bit masks.

I was thinking of something like

```python
chip.disable_all_channels()
chip.enable_channels([0, 2, 3, 4, 10])
```

Then you could use the same sort of code as in `set_pixel_trim_thresholds`.

File: python/larpix.py [hunk 2]
```diff
@@ -16,43 +16,155 @@ def __init__(self, chip_id, io_chain):
...
+        ctypes_config.test_mode = self._test_mode
+        ctypes_config.cross_trigger_mode = self._cross_trigger_mode
+        ctypes_config.periodic_reset = self._periodic_reset
+        ctypes_config.fifo_diagnostic = self._fifo_diagnostic
+        ctypes_config.sample_cycles = self._sample_cycles
+        for i,value in enumerate(self._test_burst_length):
+            ctypes_config.test_burst_length[i] = value
+        ctypes_config.adc_burst_length = self._adc_burst_length
+        for i,value in enumerate(self._channel_mask):
+            ctypes_config.channel_mask[i] = value
```
Comment (samkohn) [inline python/larpix.py]: Here you treat `self._channel_mask` as a list but you define it as a bitmap at Line 132. This is what got me mixed up.

**Comment by peter-madigan** [comment]
Yeah that's definitely a mistake. I'll fix that. What do you think about
the bit mask vs list input?

**Comment by samkohn** [comment]
I still am leaning towards lists. It makes the coding much easier since the C code already takes care of converting from a list to a bit mask. And in Python it's much more common to work with lists than bit masks .

**Comment by samkohn** [comment]
Let me know if you're working on this or if you want me to take it from here

**Comment by peter-madigan** [comment]
I'm working on it, been busy the past few days.

**Comment by samkohn** [comment]
Cool it's not a huge hurry, but I just wanted to make sure I shouldn't work on it.

**Pull request merged**